### PR TITLE
46 allow use in lib package

### PR DESCRIPTION
**Pull request by tjadevries**
Another significant refactoring, but the end result is really nice I think: Sections for singleton behavior, Containers for file handling. I hope you like it as much as I do.
And I propose to release 0.3.0 after this has been merged.

### CODE REVIEW

File: src/application_settings/container_section_base.py [hunk 1]
```diff
@@ -11,10 +12,72 @@
...
+        for subsec in subsections:
+            subsec._set()  # pylint: disable=protected-access
+        return self
+
     def _update(self, changes: dict[str, dict[str, Any]]) -> Self:
         "Update parameters and sections with data specified in changes; not meant for config"
         return replace(self, **changes)
+
+
+_ALL_CONTAINER_SECTION_SINGLETONS: dict[int, Any] = {}
```
Comment (JJdeVries) [inline src/application_settings/container_section_base.py:83]: I think we could use as a type `dict[int, ContainerSectionBase]` right, as it should contain some derived class of that.

**Review by JJdeVries — CHANGES_REQUESTED** [review]

File: src/application_settings/container_base.py [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
+from typing import Any
 
 import tomli_w
 from pathvalidate import is_valid_filepath
 from pydantic.dataclasses import dataclass
 
-from .container_section_base import ContainerSectionBase
-from .type_notation_helper import PathOpt, PathOrStr
+from application_settings.container_section_base import ContainerSectionBase
+from application_settings.type_notation_helper import PathOpt, PathOrStr
```
[the remark this answers is not in the document]
Comment (tjadevries) [reply src/application_settings/container_base.py:16]: I got errors with the relative imports when I wrote example_settings.py (don't really understand why). Then I read this: https://realpython.com/absolute-vs-relative-python-imports/
Based on that I think that absolute imports are fine (and to be preferred) here.

File: src/application_settings/container_section_base.py (hunk 1 above)
Comment (tjadevries) [reply src/application_settings/container_section_base.py:83]: Agreed, changed

**Comment by tjadevries** [comment]
I updated the dependencies, hence if this PR is merged, all dependabot PRs will disappear.

**Review by JJdeVries — APPROVED** [review]

**Pull request merged**